File: src/services/leaderboard.ts
import api from './api';

export interface LeaderboardEntry {
  userId: string;
  username: string;
  profit: number;
  rank: number;
  profitPercentage: number;
  avatar?: string;
}

export type TimeFrame = '24h' | '7d' | '30d';

export const getLeaderboard = async (timeframe: TimeFrame): Promise<LeaderboardEntry[]> => {
  const response = await api.get<LeaderboardEntry[]>(`/leaderboard?timeframe=${timeframe}`);
  return response.data;
}; 

File: internal/models/leaderboard.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type LeaderboardEntry struct {
	ID               uuid.UUID `json:"id" db:"id"`
	UserID           uuid.UUID `json:"user_id" db:"user_id"`
	Username         string    `json:"username" db:"username"`
	Profit           float64   `json:"profit" db:"profit"`
	ProfitPercentage float64   `json:"profit_percentage" db:"profit_percentage"`
	Rank             int       `json:"rank" db:"rank"`
	TimeFrame        string    `json:"timeframe" db:"timeframe"`
	CreatedAt        time.Time `json:"created_at" db:"created_at"`
}
